Readme update (#453)
brainhubeu/react-carousel: README.md

- `BrainhubeuReactCarouselItem`
- `BrainhubeuReactCarouselWrapper`

## Usage

> By default the component does not need anything except children to render simple carousel.

Reviewer: ```suggestion
By default, the component does not need anything except children to render simple carousel.
```

> Remember that styles does not have to be imported everytime you use carousel, you can do it once in an entry point of your bundle.

Reviewer: ```suggestion
Remember that styles do not have to be imported every time you use carousel, you can do it once in an entry point of your bundle.
```

```javascript
import React, { Component } from 'react';
import Carousel from '@brainhubeu/react-carousel';
import '@brainhubeu/react-carousel/lib/style.css';

export default class MyCarousel extends Component {
  render() {
    return (
      <Carousel arrows dots>
        <img src={imageOne} />
        <img src={imageTwo} />
        <img src={imageThree} />
      </Carousel>
    );
  }
}
```

[![gif](readme/assets/carousel.gif)](https://github.com/brainhubeu/react-carousel/)

### Showing dots or thumbnails

> There is a separate Dots component that can be used to fully controll navigation dots or add thumbnails.

Reviewer: ```suggestion
There is a separate Dots component that can be used to fully control navigation dots or add thumbnails.
```

}
```

[![gif](readme/assets/thumbnails.gif)](https://github.com/brainhubeu/react-carousel/)

## Props
You can access a clickable demo with many examples and a [live code editor](https://brainhubeu.github.io/react-carousel/) by clicking on a Prop name.

### Carousel props

> | Prop | Type | Description |

Reviewer: could you add a `Default` column?

| --- | --- | --- |
| [**addArrowClickHandler**](https://brainhubeu.github.io/react-carousel/docs/examples/customArrows) | *Boolean* | Has to be added for arrowLeft and arrowRight to work |
| [**animationSpeed**](https://brainhubeu.github.io/react-carousel/docs/examples/animation) | *Number* | Determines transition duration in milliseconds |

> | [**arrowLeft**](https://brainhubeu.github.io/react-carousel/docs/examples/customArrows) | *React element* | To be used instead of default left arrow (if you provide these custom arrows, you don't have to use arrows prop) |

Reviewer: ```suggestion
| [**arrowLeft**](https://brainhubeu.github.io/react-carousel/docs/examples/customArrows) | *React element* | To be used instead of the default left arrow (if you provide these custom arrows, you don't have to use arrows prop) |
```

> | [**arrowRight**](https://brainhubeu.github.io/react-carousel/docs/examples/customArrows) | *React element* | To be used instead of default right arrow (if you provide these custom arrows, you don't have to use arrows prop) |

Reviewer: ```suggestion
| [**arrowRight**](https://brainhubeu.github.io/react-carousel/docs/examples/customArrows) | *React element* | To be used instead of the default right arrow (if you provide these custom arrows, you don't have to use arrows prop) |
```

> | [**arrows**](https://brainhubeu.github.io/react-carousel/docs/examples/defaultArrows) | *Boolean* | Renders default arrows |

Reviewer: For each Boolean prop, could you formulate the description to start with the following?
```
If set to `true`, it...
```
and could you add `Otherwise, ...` when it's not obvious?

Author: I'm not sure about it. I saw many good repositories without this additional phrase and I think we don't need to extend our descriptions.

Reviewer: or even two separate columns for booleans:
- `behavior when true`
- `behavior when false`

Author: I'm sorry, this not a common solution. I think that will only add more text to read and this is redundant. If you do not agree, we can vote on Thursday's meeting.

Reviewer: I can accept the current solution for now and we could discuss this on Thursday.

| [**autoPlay**](https://brainhubeu.github.io/react-carousel/docs/examples/animation) | *Number* | Slide change interval in milliseconds |

> | [**breakpoints**](https://brainhubeu.github.io/react-carousel/docs/examples/responsive) | *Object* | All props (except value, onChange, responsive, children) can be set to different values on different screen resolution |

Reviewer: ```suggestion
| [**breakpoints**](https://brainhubeu.github.io/react-carousel/docs/examples/responsive) | *Object* | All props (except of `value`, `onChange`, `responsive`, `children`) can be set to different values on different screen resolutions |
```

| [**dots**](https://brainhubeu.github.io/react-carousel/docs/examples/simpleDots) | *Boolean* | Renders default dots under the carousel |
| [**draggable**](https://brainhubeu.github.io/react-carousel/docs/examples/draggable) | *Boolean* | Makes it possible to drag to the next slide with mouse cursor |
| [**infinite**](https://brainhubeu.github.io/react-carousel/docs/examples/infinite) | *Boolean* | Creates an infinite carousel width |
| **itemWidth** | *Number* | Determines custom width for every slide in the carousel |
| [**keepDirectionWhenDragging**](https://brainhubeu.github.io/react-carousel/docs/examples/draggable) | *Boolean* | While dragging, it doesn't matter which slide is the nearest one, but in what direction you dragged |
| **minDraggableOffset** | *Number* | Defines the minimum offset to consider the drag gesture |
| **offset** | *Number* | Padding between items |
| [**onChange**](https://brainhubeu.github.io/react-carousel/docs/examples/controlled) | *Function* | Handler triggered when current slide is about to change (e.g. on arrow click or on swipe) |

> | [**rtl**](https://brainhubeu.github.io/react-carousel/docs/examples/rtl) | *Boolean* | Iindicating if the carousel should have direction from Right to Left (make sure to pass the rtl param to the Dots component as well) |

Reviewer: ```suggestion
| [**rtl**](https://brainhubeu.github.io/react-carousel/docs/examples/rtl) | *Boolean* | Indicating if the carousel should have direction from Right to Left (make sure to pass the `rtl` param to the `Dots` component as well) |
```

| **slides** | *Array* | Alternative way to pass slides. This prop expects an array of JSX <img> elements |
| [**slidesPerPage**](https://brainhubeu.github.io/react-carousel/docs/examples/multipleItems) | *Number* | Number of slides visible at once |
| [**slidesPerScroll**](https://brainhubeu.github.io/react-carousel/docs/examples/slidesPerScroll) | *Number* | Number by which value will change on scroll (autoPlay, arrow click, drag)|
| **stopAutoPlayOnHover** | *Boolean* | Determines if autoPlay should stop when mouse hover over carousel |
| [**value**](https://brainhubeu.github.io/react-carousel/docs/examples/controlled) | *Number* | Current slide's index (zero based, depends on the elements order) |

### Dots props

| Prop | Type | Description |
| --- | --- | --- |

> | [**number**](https://brainhubeu.github.io/react-carousel/docs/examples/dots) | *Number* | Number of slides in carousel you want to control. By default it will be set to the number of thumbnails (if provided) |

Reviewer: ```suggestion
| [**number**](https://brainhubeu.github.io/react-carousel/docs/examples/dots) | *Number* | Number of slides in the carousel you want to control. By default, it will be set to the number of thumbnails (if provided) |
```

> | [**onChange**](https://brainhubeu.github.io/react-carousel/docs/examples/dots) | *Function* | OnChange callback (works the same way as onChange in Carousel component) |

Reviewer: ```suggestion
| [**onChange**](https://brainhubeu.github.io/react-carousel/docs/examples/dots) | *Function* | `OnChange` callback (works the same way as `onChange` in `Carousel` component) |
```

| **rtl** | *Boolean* | Indicating if the dots should have direction from Right to Left |
| [**thumbnails**](https://brainhubeu.github.io/react-carousel/docs/examples/thumbnails) | *Array of ReactElements* | Array of thumbnails to show. If not provided, default dots will be shown |

> | [**value**](https://brainhubeu.github.io/react-carousel/docs/examples/dots) | *Number* | Current Carousel value |

Reviewer: ```suggestion
| [**value**](https://brainhubeu.github.io/react-carousel/docs/examples/dots) | *Number* | Current `Carousel` value |
```

## Tests

### Unit tests
```
yarn test:unit
```

### E2E tests
```
yarn test:e2e
```

## Contributing
[The GitHub issues list](https://github.com/brainhubeu/react-carousel/issues) is our roadmap.
You're more than welcome to vote:
- with 👍if you like a given feature request or you'd like a given bug to be fixed
- with ❤️ if you love a given feature request or fixing a given bug is critical for you
- with 👎if in your opinion, a given feature would create more damages than the value provided by it or you consider a given bug to be a feature

We don't give any guarantee to fix even the most liked issues but 👍and ❤️ increase probability of fixing while 👎decreases the probability of fixing.

You're also more than welcome to:
- submit a feature request
- report a bug
- ask a question
- comment an issue, discussing the details
- open a PR, fixing a given issue

## Decision log

### SSR

> When using `@brainhubeu/react-carousel` with SSR (Server-side Rendering), we recommend [Next.js](https://github.com/zeit/next.js) as `@brainhubeu/react-carousel` currently doesn't work on the server side so it must be rendered on the client side (maybe we'll provide server-side working in the future).

Reviewer: other decisions:
- `cypress` instead of `gemini`/`hermione`
- `renovate` instead of `greenkeeper`
- deploying a test environment for each PR to `beghp`
- IssueHunt?

Author: We decided to move it to separate ticket

```js
import dynamic from 'next/dynamic';

const { default: Carousel, Dots } = dynamic(
 () => require('@brainhubeu/react-carousel'),
 { ssr: false },
);
```

## License

> react-carousel is copyright © 2018-2020 [Brainhub](https://brainhub.eu/?utm_source=github) It is free software, and may be redistributed under the terms specified in the [license](LICENSE.md).

Reviewer: ```suggestion
react-carousel is copyright © 2018-2020 [Brainhub](https://brainhub.eu/?utm_source=github). It is free software, and may be redistributed under the terms specified in the [license](LICENSE.md).
```

| [**keepDirectionWhenDragging**](https://brainhubeu.github.io/react-carousel/docs/examples/draggable) | *Boolean* | `undefined` | While dragging, it doesn't matter which slide is the nearest one, but in what direction you dragged |
| **minDraggableOffset** | *Number* | `10` | Defines the minimum offset to consider the drag gesture |
| **offset** | *Number* | `0` | Padding between items |
| [**onChange**](https://brainhubeu.github.io/react-carousel/docs/examples/controlled) | *Function* | `undefined` | Handler triggered when current slide is about to change (e.g. on arrow click or on swipe) |
| [**rtl**](https://brainhubeu.github.io/react-carousel/docs/examples/rtl) | *Boolean* | `false` | Indicating if the carousel should have direction from Right to Left (make sure to pass the `rtl` param to the `Dots` component as well) |
| **slides** | *Array* | `undefined` | Alternative way to pass slides. This prop expects an array of JSX <img> elements |
| [**slidesPerPage**](https://brainhubeu.github.io/react-carousel/docs/examples/multipleItems) | *Number* | `1` | Number of slides visible at once |

> | [**slidesPerScroll**](https://brainhubeu.github.io/react-carousel/docs/examples/slidesPerScroll) | *Number* | `1` | śNumber by which value will change on scroll (autoPlay, arrow click, drag)|

Reviewer: ```suggestion
| [**slidesPerScroll**](https://brainhubeu.github.io/react-carousel/docs/examples/slidesPerScroll) | *Number* | `1` | Number by which value will change on scroll (autoPlay, arrow click, drag)|
```
